sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/HashExpressionsSuite.scala

```diff
@@ -71,6 +75,242 @@ class HashExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper {
     checkConsistencyBetweenInterpretedAndCodegen(Crc32, BinaryType)
   }
 
+
+  def checkHiveHash(value: Any, dataType: DataType, expected: Long): Unit = {
+    // Note : All expected hashes need to be computed using Hive 1.2.1
+    val actual = HiveHashFunction.hash(value, dataType, seed = 0)
+    assert(actual == expected)
```

Comment:
we should add a clue; otherwise we will never be able to tell what's going on if the tests fail on those randomized vlaues.

```
withClue(s"value is $value") {
  assert(..
}
```